diff --git a/node_modules/@creamy-dev/qbuild/index.js b/node_modules/@creamy-dev/qbuild/index.js
index c7da6ee..eb0288a 100644
--- a/node_modules/@creamy-dev/qbuild/index.js
+++ b/node_modules/@creamy-dev/qbuild/index.js
@@ -1,19 +1,13 @@
-const os = require("os");
 const fs = require("fs");
+const path = require("path");
 
 function stringParser(str) {
   return str.split(str[0])[1];
 }
 
-function compile(path, disableWelcome) {
-  const delimiter = os.platform() == "win32" ? "\\" : "/";
-
-  const rootPathArr = path.split(delimiter);
-  rootPathArr.pop();
-
-  const rootPath = rootPathArr.join(delimiter);
-
-  const file = fs.readFileSync(path, "utf-8");
+function compile(origPath, disableWelcome) {
+  const rootPath = path.join(origPath, "..");
+  const file = fs.readFileSync(origPath, "utf-8");
 
   const fileSplit = file.split("\n");
   let isRegularRequireEnabled = false;
@@ -31,9 +25,9 @@ function compile(path, disableWelcome) {
       const rawCommand = newDelimiter + trim.split(newDelimiter)[1].split(")")[0] + ")";
       
       const line = rawCommand.split("(")[1].split(")")[0];
-      const path = stringParser(line).replace("./", "/");
+      const parsedPath = stringParser(line).replace("./", "/");
 
-      const newPath = rootPath + path;
+      const newPath = path.join(rootPath, parsedPath);
       
       const compiled = file.replace(rawCommand, compile(newPath, true));
       newLines.push(compiled);
